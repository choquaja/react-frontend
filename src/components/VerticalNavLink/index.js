import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { themeColor } from '../../services/theme';

const VerticalNavLink = styled(NavLink).attrs({
  activeClassName: 'active',
})`
  display: block;
  padding: 1.2rem .8rem;
  font-size: 1.1em;
  color: inherit;
  text-decoration: none;
  &:hover {
    background-color: ${themeColor('gray2')};
  }
  &[class~="active"] {
    background-color: ${themeColor('primary')};
    color: ${themeColor('white')};
  }
`;

export default VerticalNavLink;
