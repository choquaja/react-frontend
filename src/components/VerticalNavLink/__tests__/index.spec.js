import React from 'react';
import { shallow } from 'enzyme';
import VerticalNavLink from '../';

const required = {
  to: 'test',
};
describe('VerticalNavLink', () => {
  it('renders without crashing', () => {
    shallow(<VerticalNavLink {...required} />);
  });

  it('renders with to string prop', () => {
    const component = shallow(<VerticalNavLink {...required} />);
    expect(component).toMatchSnapshot();
  });
});
