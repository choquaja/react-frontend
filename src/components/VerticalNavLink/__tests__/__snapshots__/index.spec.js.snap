// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalNavLink renders with to string prop 1`] = `
<NavLink
  activeClassName="active"
  className="VerticalNavLink__VerticalNavLink-dLQMAE fIsCYa"
  to="test"
/>
`;
